### PR TITLE
chore(tests): add tests for collectors

### PR DESCRIPTION
**Pull request by countless-integers**
tests(unit.collectors): fix command collector test
tests(unit.collector): stub for a framework collector test

### CODE REVIEW

File: tests/unit/Collectors/CommandCollectorTest.php [hunk 1]
```diff
@@ -111,16 +110,48 @@ public function testCommandFinishedListenerIgnored(): void
...
+        $this->dispatcher->dispatch(
+            new CommandStarting(
+                self::COMMAND_NAME,
+                $this->commandInputMock,
+                $this->commandOutputMock
+            ));
+    }
+
+    /**
+     * @group shit
```
Comment (arkaitzgarro) [inline tests/unit/Collectors/CommandCollectorTest.php]: WAT

**Comment by arkaitzgarro** [comment]
@countless-integers I added unit tests for all collectors but `ScheduledTaskCollector`. The events used by this collector are not available in Laravel 5.5 (this is the default version we install for development) and the tests will fail. It would be really nice if we could install different versions of Laravel and run the suite for each version...

File: tests/unit/Collectors/FrameworkCollectorTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public function testCollectorName(): void
+    {
+        self::assertEquals('framework-collector', $this->collector->getName());
+    }
+
+    public function testItCanRegisterBootingEvent(): void
+    {
+        $this->markTestSkipped('Depending on the order tests are executed, Laravel boot event ');
+
+        $this->app->boot();
```
Comment (arkaitzgarro) [inline tests/unit/Collectors/FrameworkCollectorTest.php:34]: Not sure how (I'm assuming `app(Application::class)` follows the singleton pattern), but depending on the order the tests are executed, the application is not [firing the callbacks](https://github.com/laravel/framework/blob/5.5/src/Illuminate/Foundation/Application.php#L765).

Cannot we make sure `$this->app` is always a new fresh instance?

Comment (countless-integers) [reply tests/unit/Collectors/FrameworkCollectorTest.php:34]: I've changed it to create a new instance, but I cannot really explain why it was failing before. [Also removed setUp as  it's not a codeception method](https://stackoverflow.com/questions/41934684/what-is-the-difference-between-setup-in-phpunit-and-before-in-codeception), but a phpunit one, so to remove any confusion about when that is run.

File: tests/bootstrap.php [hunk 3]
```diff
@@ -1,3 +1,5 @@
 <?php
 
+define('LARAVEL_START', 0);
```
Comment (countless-integers) [inline tests/bootstrap.php:3]: 👍 yep, that's a better way of doing it 😉

File: tests/unit/Collectors/DBQueryCollectorTest.php [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    /** @var Connection|LegacyMockInterface|MockInterface */
+    private $connectionMock;
+
+    protected function _before(): void
+    {
+        $this->app = app(Application::class);
+        $this->dispatcher = app(Dispatcher::class);
+    }
+
+    protected function setUp(): void
```
Comment (countless-integers) [inline tests/unit/Collectors/DBQueryCollectorTest.php]: why not just use `_before` and `_after` ?

File: tests/unit/Collectors/DBQueryCollectorTest.php [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        self::assertGreaterThan(0.0, $measure['start']);
+        self::assertEquals(500.0, $measure['duration']);
+        self::assertEquals('db.mysql.query', $measure['type']);
+        self::assertEquals('query', $measure['action']);
+    }
+
+    public function testDisabledFastQueryExecutedListener(): void
+    {
+        $this->configMock
+            ->shouldReceive('get')
```
Comment (countless-integers) [inline tests/unit/Collectors/DBQueryCollectorTest.php]: if it's meant to be called once then:

```suggestion
            ->expects('get')
```

File: tests/unit/Collectors/DBQueryCollectorTest.php [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            ->with('elastic-apm-laravel.spans.querylog.enabled')
+            ->andReturn('auto');
+
+        $this->configMock
+            ->shouldReceive('get')
```
Comment (countless-integers) [inline tests/unit/Collectors/DBQueryCollectorTest.php]:
```suggestion
            ->expects('get')
```

File: tests/unit/Collectors/DBQueryCollectorTest.php [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+            new QueryExecuted(
+                self::SQL,
+                [1],
+                500,
+                $this->connectionMock
+            )
+        );
+
+        $measure = $this->collector->collect()->first();
+        self::assertNull($measure);
```
Comment (countless-integers) [inline tests/unit/Collectors/DBQueryCollectorTest.php]: would that also mean that `self::assertTrue($this->collector->collect()->isEmpty();`?

File: tests/unit/Collectors/DBQueryCollectorTest.php [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    public function testFallbackQueryName(): void
+    {
+        $this->configMock
+            ->shouldReceive('get')
```
Comment (countless-integers) [inline tests/unit/Collectors/DBQueryCollectorTest.php]:
```suggestion
            ->expects('get')
```

File: tests/unit/Collectors/ScheduledTaskCollectorTest.php [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+            $this->app,
+            new Config([]),
+            new RequestStartTime(0.0)
+        );
+        $this->collector->useAgent(Mockery::mock(Agent::class));
+    }
+
+    public function testCollectorName(): void
+    {
+        self::assertEquals('scheduled-task-collector', $this->collector->getName());
```
Comment (countless-integers) [inline tests/unit/Collectors/ScheduledTaskCollectorTest.php:37]: 😉

File: tests/unit/Exception/MissingAppConfigurationExceptionTest.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    private $exception;
+
+    protected function _before(): void
+    {
+        $this->exception = new MissingAppConfigurationException();
+    }
+
+    public function testExceptionMessage(): void
+    {
+        self::assertEquals('Use AgentBuilder::withAppConfig() to provide a configuration object.', $this->exception->getMessage());
```
Comment (countless-integers) [inline tests/unit/Exception/MissingAppConfigurationExceptionTest.php:18]: I'm surprised linter doesn't complain about line length.

File: tests/unit/Collectors/FrameworkCollectorTest.php (hunk 2 above)
Comment (arkaitzgarro) [reply tests/unit/Collectors/FrameworkCollectorTest.php:34]: > I've changed it to create a new instance, but I cannot really explain why it was failing before.

Nice, thanks!

>[Also removed setUp as it's not a codeception method](https://stackoverflow.com/questions/41934684/what-is-the-difference-between-setup-in-phpunit-and-before-in-codeception), but a phpunit one, so to remove any confusion about when that is run.

I'm going to update the rest of tests that are using `setup/teardown` as well.

File: tests/unit/Collectors/DBQueryCollectorTest.php (hunk 4 above)
Comment (arkaitzgarro) [reply tests/unit/Collectors/DBQueryCollectorTest.php]: Yes, I'm going to update all methods to be `_before` and `_after`.

File: tests/unit/Collectors/DBQueryCollectorTest.php (hunk 7 above)
Comment (arkaitzgarro) [reply tests/unit/Collectors/DBQueryCollectorTest.php]: This is a cleaner assert, I will change it.

File: tests/unit/Exception/MissingAppConfigurationExceptionTest.php (hunk 10 above)
Comment (arkaitzgarro) [reply tests/unit/Exception/MissingAppConfigurationExceptionTest.php:18]: [From PSR-2](https://www.php-fig.org/psr/psr-2/):

>There MUST NOT be a hard limit on line length; the soft limit MUST be 120 characters; lines SHOULD be 80 characters or less.

I love the contradiction, no hard limit but soft limit MUST be.

**Review by arkaitzgarro — APPROVED** [review]

**Pull request merged**